store/src/db/sql.rs

```diff
@@ -399,6 +400,48 @@ pub fn select_notes_since_block_by_tag_and_sender(
     Ok(res)
 }
 
+/// Select Note's matching the NoteId using the given [Connection].
+///
+/// # Returns
+///
+/// - Empty vector if no matching `note`.
+/// - Otherwise, notes which `note_hash` matches the `NoteId` as bytes.
+pub fn select_notes_by_id(
+    conn: &mut Connection,
+    note_ids: &[NoteId],
+) -> Result<Vec<Note>> {
+    let note_ids: Vec<Vec<u8>> = note_ids.iter().map(|id| id.to_bytes()).collect();
+
+    let placeholders = note_ids.iter().map(|_| "?").collect::<Vec<&str>>().join(", ");
+
+    let query = &format!("SELECT * FROM notes WHERE note_hash IN ({})", placeholders);
```

Comment:
Instead of formatting the SQL statement, this should use the `rarray` extension. Here is an example:

https://github.com/0xPolygonMiden/miden-node/blob/0335339c6bdaf3daf07dd3750e2a19b97e47ba66/store/src/db/sql.rs#L83-L96

In general I'm against formatting SQL statements, for two reasons:

- It is an easy way to get into a SQL injection vulnerability
- It makes it harder to have caches of prepared statements

Reply:
> The `note_hash` column name should likely be updated to `note_id` but not for this set of changes

Opening an issue here: https://github.com/0xPolygonMiden/miden-node/issues/301